vpr/SRC/route/route_timing.c

```diff
@@ -17,14 +18,31 @@
 #include "net_delay.h"
 #include "stats.h"
 #include "ReadOptions.h"
-
-
+ 
 using namespace std;
 
+/***************** Iterative connection based rerouting **********************/
 
+// conceptually works like rr_node_to_pin[net_index][sink_node_index] to get the pin index for that net
+// each net maps SINK node index -> PIN index for net
+// only need to be built once at the start since the SINK nodes never change
+// the reverse lookup of net_rr_terminals
+// do not query this directly, as rr_node_to_pin[inet][inode] will create the entry if it does not exist
+// instead, use it through functions convert_sink_node_to_pins_of_net and put_sink_rt_nodes_to_pins_lookup
+static vector<unordered_map<int,int>> rr_node_to_pin;
+
+// a property of each net, but only valid after pruning the previous route tree
+static vector<int> remaining_targets;
+static vector<t_rt_node*> reached_sinks;	// used to populate rt_node_of_sink after building route tree from traceback

```

Comment:
Consider putting above file-scope variables into a class and include functions that initialize/update these variables as class methods. If it's not messy, consider moving file-scope variables (on their own or as part of a class) down into functions -- local variables are easy to track back up the function hierarchy; on the other hand i often have trouble following file-scope/globals. If file-scope variables remain, prepend f_ to their name.

Variable comments:
remaining_targets and reached_sinks are good names, but explicit comments mentioning what the variables contain would be useful (e.g. sinks that have been reached while routing current net)